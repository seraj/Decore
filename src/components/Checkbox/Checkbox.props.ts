import InputProps from "../Input/Input.props";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export default interface CheckboxProps extends InputProps {}
