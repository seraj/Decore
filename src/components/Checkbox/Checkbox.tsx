import * as React from "react";
import StyledCheckbox, {
  HiddenCheckbox,
  CheckboxContainer
} from "./styled/StyledCheckbox";
import CheckboxProps from "./Checkbox.props";
import Icon from "../Icon";
import InputProps from "../Input/Input.props";

const Checkbox: React.FC<CheckboxProps> = ({
  checked,
  disabled,
  className,
  children,
  ...props
}) => (
  <CheckboxContainer className={className}>
    <HiddenCheckbox
      {...props}
      type="checkbox"
      checked={checked}
      disabled={disabled}
    />
    <StyledCheckbox checked={checked} disabled={disabled}>
      <Icon
        src={
          "data:image/svg+xml,%3Csvg width='10' height='8' viewBox='0 0 10 8' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M9.87216 0.936324L9.06101 0.058217C9.02874 0.0194056 8.97805 0 8.93196 0C8.88126 0 8.83517 0.0194056 8.80291 0.058217L3.18011 6.02062L1.13378 3.86659C1.09691 3.82777 1.05082 3.80837 1.00473 3.80837C0.958641 3.80837 0.912552 3.82777 0.875682 3.86659L0.0553062 4.73014C-0.0184354 4.80776 -0.0184354 4.92905 0.0553062 5.00667L2.63626 7.72347C2.80218 7.89812 3.00497 8 3.1755 8C3.41977 8 3.63178 7.81079 3.71013 7.73317C3.71013 7.73317 3.71013 7.73317 3.71474 7.73317L9.87677 1.21286C9.9413 1.13038 9.9413 1.0091 9.87216 0.936324Z' fill='%23333333'/%3E%3C/svg%3E"
        }
      ></Icon>
    </StyledCheckbox>
  </CheckboxContainer>
);

export default Checkbox;
