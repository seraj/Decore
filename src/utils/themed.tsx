import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
