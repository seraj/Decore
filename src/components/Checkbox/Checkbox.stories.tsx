import { withKnobs, boolean } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Checkbox from "./Checkbox";

const stories = storiesOf("Components/Checkbox", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <Checkbox
      checked={boolean("checked", false)}
      disabled={boolean("disabled", false)}
    />
  ))
);
