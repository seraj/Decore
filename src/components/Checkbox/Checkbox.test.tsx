import * as React from "react";

import themed from "../../utils/themed";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
  it("should render correctly", () => {
    const component = themed(<Checkbox />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
