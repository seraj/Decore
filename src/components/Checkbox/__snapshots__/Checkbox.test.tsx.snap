// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
.c3 {
  border: 1.5px solid #999999;
  background-color: undefined;
  padding: 1rem 0.8rem;
  height: 45px;
  border-radius: 0px;
  color: #333333;
}

.c3:hover {
  border-color: #333333;
}

.c3:disabled {
  background: #E5E5E5;
  color: #999999;
  cursor: not-allowed !important;
  pointer-events: none;
}

.c5 {
  display: inline-block;
  fill: #333333;
  width: 12px;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clippath: inset(50%);
  clippath: inset(50%);
  height: 1px !important;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px !important;
}

.c4 {
  display: inline-block;
  width: 20px;
  height: 20px;
  background: #ffffff;
  border-color: #999999;
  border-width: 1.5px;
  border-style: solid;
}

.c4:hover {
  border-color: #666666;
}

.c4[disabled] {
  background: #E5E5E5;
  border-color: #E5E5E5;
  pointer-events: none;
}

.c4[disabled] svg * {
  fill: #999999;
}

.c1:focus + .c4 {
  box-shadow: 0 0 0 3px pink;
}

.c4 function (_a) {
  -webkit-var: children = _a.children,props = __rest(_a,["children"]);
  -moz-var: children = _a.children,props = __rest(_a,["children"]);
  -ms-var: children = _a.children,props = __rest(_a,["children"]);
  var: children = _a.children,props = __rest(_a,["children"]);
  return: (React.createElement(StyledIcon_1.default,__assign({},props),children));
}

.c4 {
  position: relative;
}

.c4 svg {
  visibility: hidden;
  position: absolute;
  top: 5px;
  left: 3px;
}

<div
  className="c0"
>
  <input
    className="c1 c2 c3"
    type="checkbox"
  />
  <div
    className="c4"
  >
    <span
      className="isvg loaded c5"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg width='10' height='8' viewBox='0 0 10 8' fill='none' xmlns='http://www.w3.org/2000/svg'><path d='M9.87216 0.936324L9.06101 0.058217C9.02874 0.0194056 8.97805 0 8.93196 0C8.88126 0 8.83517 0.0194056 8.80291 0.058217L3.18011 6.02062L1.13378 3.86659C1.09691 3.82777 1.05082 3.80837 1.00473 3.80837C0.958641 3.80837 0.912552 3.82777 0.875682 3.86659L0.0553062 4.73014C-0.0184354 4.80776 -0.0184354 4.92905 0.0553062 5.00667L2.63626 7.72347C2.80218 7.89812 3.00497 8 3.1755 8C3.41977 8 3.63178 7.81079 3.71013 7.73317C3.71013 7.73317 3.71013 7.73317 3.71474 7.73317L9.87677 1.21286C9.9413 1.13038 9.9413 1.0091 9.87216 0.936324Z' fill='#333333'/></svg>",
        }
      }
    />
  </div>
</div>
`;
